### PR TITLE
Unit test commit cmd

### PR DESCRIPTION
**Pull request by bpeabody**
Unit tests for the commit and commit --amend commands.

### CODE REVIEW

**Comment by bpeabody** [comment]
addresses https://github.com/twosigma/git-meta/issues/283

File: node/lib/util/commit.js [hunk 1]
```diff
@@ -1762,3 +1767,280 @@ exports.parseSplitCommitMessages = function (text) {
...
+configuration changes.`);
+    }
+}
+
+function abortForNoMessage() {
+    throw new UserError("Aborting commit due to empty commit message.");
+}
+
+/**
+ * Perform the commit command in the specified `repo`.  Use the specified `cwd`
```
Comment (novalis) [inline node/lib/util/commit.js]: Can you clarify how `cwd` is used?  As in, "paths are displayed relative to `cwd`" or whatever.

**Comment by novalis** [comment]
still one failing test...

**Comment by bpeabody** [comment]
I think these problems are with travis. The places where it's dying are different every time: weird places where I've never seen a test fail before -- this latest one in my stopwatch class!

I will keep pushing new commits until it succeeds.

**Comment by bpeabody** [comment]
Finally!

**Pull request merged**